### DISCUSSION TITLE
One image per git repo

### DISCUSSION THREAD

**Discussion by lmilbaum** [CentOS/centos-bootc#89] [Q&A] [answered]
This repo produces two container images.  It would be preferable if it would produce only one container image while the second one is tracked in its own repo. 
Eventually, the target is to migrate the fedora one to be owned by fedora so it makes sense to split the sources at some point.
As another argument, it is required to define those deliverables (container images) in different RHTAP applications due to tagging constrains. 
Wouldn't it make sense to perform the split in conjunction with the applications splitting?

**Comment by cgwalters** [comment]
A lot hinges on the definition of "split".  If it's literally a fork, that will impose a cost.  Let's be realistic, right now this repository is just a list of packages, but still, there's already too many of these repositories floating around (atomic desktops, edge, coreos etc. and a big goal of this project is to deduplicate).

**Reply by cgwalters** [reply]
One approach is to use git submodules.  That's what https://github.com/coreos/fedora-coreos-config/pull/2610 is doing.

I guess the question of whether centos-bootc git-submodules fedora-bootc or there's a common-bootc git that is git-submoduled by both of them...

Or we try to do just do this "manually" with CI/automation that automatically syncs files from one place to another.

**Reply by lmilbaum** [reply]
Would handling the relationship of those two repos (fedora-bootc and centos-bootc) as upstream and downstream can address your concern?
The idea behind it is that the centos-bootc repo will not duplicate the sources but track a git commit sha to the fedora-bootc repo and the specific files which are centos specific.

**Reply by cgwalters** [reply]
Ultimately again we clearly want to move the Fedora one in some place "upstream".  And longer term, the build process for it too (which is kind of happening in https://pagure.io/pungi/pull-request/1699 ).

I am just questioning whether at this very moment it's *really* necessary to split the git repositories here.

**Reply by lmilbaum** [reply]
It is not really necessary. It is just a matter of saving some cycles in the near future.

**Reply by cgwalters** [reply]
OK then I think it'd be best to try to defer any more "internal" restructuring in favor of having things in a more final location.

If it helps I guess for now, the focus of this repository should probably move to stream9, but we should at least try to keep building eln.

**Reply by lmilbaum** [reply, marked as answer]
Got it. Please ignore the fedora-bootc repo for now.
